### PR TITLE
Replace the OpenStack CCM Go template with an embedded addon

### PR DESCRIPTION
**Pull request by xmudrii**
**What this PR does / why we need it**:

This PR replaces the OpenStack CCM Go template with an embedded addon.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
xref #1357

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @kron4eg

### CODE REVIEW

**Comment by xmudrii** [comment]
/hold
to be manually tested

File: pkg/templates/externalccm/openstack.go [hunk 1]
```diff
@@ -1,279 +0,0 @@
...
-		return errors.New("cloudConfig not defined")
-	}
-
-	ctx := context.Background()
-
-	sa := osServiceAccount()
-	ccmRole := osCCMClusterRole()
-	ccmImage := s.Images.Get(images.OpenstackCCM)
-	ds := osDaemonSet(ccmImage)
-	cabundle.Inject(s.Cluster.CABundle, &ds.Spec.Template)
```
Comment (xmudrii) [inline pkg/templates/externalccm/openstack.go:58]: The CA bundle inject is missing in the new manifest.

**Review by kron4eg — APPROVED** [review]

**Comment by xmudrii** [comment]
@kron4eg PTAL
/hold cancel

File: pkg/addons/manifest.go [hunk 2]
```diff
@@ -185,7 +187,7 @@ func combineManifests(manifests []*bytes.Buffer) *bytes.Buffer {
 	return bytes.NewBufferString(strings.Join(parts, "\n---\n") + "\n")
 }
 
-func txtFuncMap(overwriteRegistry string) template.FuncMap {
+func txtFuncMap(overwriteRegistry string, caBundle bool) template.FuncMap {
```
Comment (kron4eg) [inline pkg/addons/manifest.go:190]:
```suggestion
func txtFuncMap(overwriteRegistry string) template.FuncMap {
```

Let's avoid using such bools. Instead, let's use {{ if .Config.caBundle }} directly in template.

Comment (xmudrii) [reply pkg/addons/manifest.go]: @kron4eg Reverted that change, PTAL.

**Comment by xmudrii** [comment]
/hold
for manual testing

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**